### PR TITLE
Add EIP: Multi-edition NFT Distribution

### PR DESCRIPTION
**Pull request by henrywfyeung**
When opening a pull request to submit a new EIP, please use the suggested template: https://github.com/ethereum/EIPs/blob/master/eip-template.md

We have a GitHub bot that automatically merges some PRs. It will merge yours immediately if certain criteria are met:

 - The PR edits only existing draft PRs.
 - The build passes.
 - Your GitHub username or email address is listed in the 'author' header of all affected PRs, inside <triangular brackets>.
 - If matching on email address, the email address is the one publicly listed on your GitHub profile.

### CODE REVIEW

**Review by Bushido154 — APPROVED** [review]

File: EIPS/eip-distributor.md [hunk 1]
```diff
@@ -0,0 +1,220 @@
+---
+eip: 9527
```
Comment (abcoathup) [inline EIPS/eip-distributor.md:2]:
```suggestion
eip: 7536
```
Assigning next sequential EIP/ERC number.
EIP/ERC numbering changed to sequential from 7500 and is no longer the PR number.

Please also update the filename.

**Review by Pandapip1 — CHANGES_REQUESTED** [review]
ERCs are referenced ERC-XXXX, not `ERCXXXX`, and your multiline code blocks are missing the language

File: EIPS/eip-7536.md [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+
+## Abstract
+
+This standard is an extension of [ERC-721](./eip-721.md), which enables any ERC-721 compliant tokens, to conditionally permit the minting of various editions of child tokens with specific privileges attached.
+
+Edition is defined as a version of child tokens that encapsulates a descriptor to the parent, the address to a validator that validates rules before minting the child token, and a set of actions that can be invoked upon obtaining the child token.
+
+Each parent token can create multiple editions, each with a different set of rules to obtain the child tokens, and with different levels of privileges attached in the form of permitted actions.
+
+Upon fulfilling the corresponding rules from a specific edition, one can obtain the child token and will be able to use the token within the boundaries set by the parent token holder.
```
Comment (Pandapip1) [inline EIPS/eip-7536.md:22]:
```suggestion
This standard extends [ERC-721](./eip-721.md) to enable conditional minting of various editions of child tokens with specific privileges attached. An edition is defined as a version of a child token that encapsulates a descriptor to the parent, the address to a validator that validates rules before minting the child token, and a set of actions that can be invoked upon obtaining the child token. Each parent token can create multiple editions, each with a different set of rules to obtain the child tokens, and with different levels of privileges attached in the form of permitted actions. Upon fulfilling the corresponding rules from a specific edition, one can obtain the child token and will be able to use the token within the boundaries set by the parent token holder.
```

File: EIPS/eip-7536.md [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+
+![Parent Child and Edition Relationship Diagram](../assets/eip-7536/assets/edition.png)
```
Comment (Pandapip1) [inline EIPS/eip-7536.md:24]: Does not belong in Abstract. Abstracts should be text-only.

File: EIPS/eip-7536.md [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+
+## Motivation
+
+Most community-based relationships can be regarded as a social structure with a community owner, and multiple tiers of community members. Each tier has its own cost and benefit of joining. This can apply to the Profile-Follower relationship, Creation-Collection relationship, Producer-Subscriber relationship, Dao-Member relationship, etc.
+
+This standard gives the interfaces for implementing such a tiered system. It empowers any ERC-721 tokens with the ability to create editions of child tokens that give privileges at a cost.
```
Comment (Pandapip1) [inline EIPS/eip-7536.md:30]:
```suggestion
```
This is implied

File: EIPS/eip-7536.md [hunk 5]
```diff
@@ -0,0 +1,226 @@
...
+- Partial Copyright Transfer: This standard enables Creators to conditionally delegate different levels of copyrights, i.e. the right to produce derivative work, to the Collectors. There is no need to sell the original copy, i.e. creator token, in the market. The Creator can instead keep the token as proof of authorship, and the key to managing copy issuance.
+
+People with the following use cases can consider applying this standard:
+
+- Creators of any unique Art/Music NFTs can use this standard to sell copies to audiences. With this standard, they can retain some control over the copies.
+- Artists can use this standard to sell time-limited copies of their artwork to other artists with a copyright statement that enables the production of derivative work
+- Universities can create Graduation Certificates as NFTs and use this standard to mint a batch of non-transferable issues to their students. The Univerity retains the right to revoke any issued certificates, through action.
+
+## Rationale
```
Comment (Pandapip1) [inline EIPS/eip-7536.md:172]: Usage belongs more in Rationale than Spec, and the wording can be significantly improved

```suggestion
## Rationale

### Usage

The Distributor Interface enables any ERC-721 token to create multiple editions, each with distinct rules validated by the Validator Interface before minting a child token, and a set of actions available post-minting. This is advantageous for community building, with diverse use cases like:

- **Copy Issuance of Unique Artwork/Content:** Artists can issue multiple copies of their unique artworks with different functions and rules, providing flexibility to both creators and collectors.
- **Partial Copyright Transfer:** Creators can conditionally delegate varying levels of copyrights, allowing for derivative work production without selling the original copy.

Consider using this standard for:

- Selling copies of unique Art/Music NFTs while retaining control over the copies.
- Selling time-limited copies of artwork with attached copyright statements for derivative work production.
- Issuing non-transferable Graduation Certificates as NFTs, where the university retains the revocation right.
```

File: EIPS/eip-7536.md [hunk 6]
```diff
@@ -0,0 +1,226 @@
...
+     *
+     * @param editionHash the hash of the edition
+     * @param isPaused the state of the edition
+     */ 
+    function pauseEdition(
+        bytes32 editionHash,
+        bool isPaused
+    ) external;
+
+}
```
Comment (Pandapip1) [inline EIPS/eip-7536.md:109]:
```suggestion
pragma solidity >=0.8.0 <0.9.0;

/**
 * @notice This interface enables an [ERC-721](./eip-721.md) compliant contract to mint child tokens. A contract implementing this interface SHOULD allow any ERC-721 compliant parent token to create editions. Each edition SHOULD contain an address to a contract implementing the Validator Interface for validating minting rules, record the contract address and the tokenId of the parent token, and specify a set of permissible actions via uint96 flags.
 */
interface IDistributor {

    /**
     * @dev Emitted when a new edition is created.
     * @param editionHash The hash of the edition configuration.
     * @param tokenContract The token contract of the NFT descriptor.
     * @param tokenId The token id of the NFT descriptor.
     * @param validator The address of the validator contract.
     * @param actions The permitted functions in the descriptor contract.
     */
    event SetEdition(bytes32 editionHash, address tokenContract, uint256 tokenId, address validator, uint96 actions);
    
    /**
     * @dev Emitted when an edition is paused or resumed.
     * @param editionHash The hash of the edition configuration.
     * @param isPaused The state of the edition.
     */
    event PauseEdition(bytes32 editionHash, bool isPaused);

    /**
     * @dev Allows the parent token holder to set an edition, enabling others to mint child tokens given certain conditions.
     * @param tokenContract The token contract of the Parent token.
     * @param tokenId The token id of the Parent token.
     * @param validator The address of the validator contract.
     * @param actions The permitted functions in the descriptor contract.
     * @param initData Data input into the validator contract for setting up rules; may encode additional edition parameters.
     * @return editionHash Returns the hash of the edition configuration.
     */
    function setEdition(
        address tokenContract,
        uint256 tokenId,
        address validator,
        uint96  actions,
        bytes calldata initData
    ) external returns (bytes32 editionHash);
    
    /**
     * @dev Allows the parent token holder to pause or resume an edition.
     * @param editionHash The hash of the edition.
     * @param isPaused The state of the edition.
     */ 
    function pauseEdition(
        bytes32 editionHash,
        bool isPaused
    ) external;
}
```
Copyedited

File: EIPS/eip-7536.md [hunk 7]
```diff
@@ -0,0 +1,226 @@
...
+
+## Specification
+
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+This standard consists of a Distributor Interface and a Validator Interface. The Distributor interface MUST be implemented by a [ERC-721](./eip-721.md) compliant contract that enables the minting of child tokens. This contract SHOULD enable any ERC-721 compliant parent token to create editions. The editions SHOULD consist of an address to a contract that implements the Validator Interface, which helps to validate the rules to mint the child token. It SHOULD also record the contract address and the tokenId of the parent token, and a set of permitted actions in the form of uint96 flags. An ERC-721-compliant contract MAY implement both the Distributor Interface and the Validator Interface, or implement only the Distributor Interface and use external contracts that implement the Validator Interface.
```
Comment (Pandapip1) [inline EIPS/eip-7536.md:36]:
```suggestion
```
This is established in the interfaces themselves

File: EIPS/eip-7536.md [hunk 8]
```diff
@@ -0,0 +1,226 @@
...
+     * @param fullfilmentData the addtion data that is required for passing the validator rules
+     */
+    function validate(
+        address initiator, 
+        bytes32 editionHash,
+        uint256 conditionType,
+        bytes calldata fullfilmentData
+    ) external payable;
+
+}
```
Comment (Pandapip1) [inline EIPS/eip-7536.md:156]:
```suggestion
pragma solidity >=0.8.0 <0.9.0;

/**
 * @notice The Validator Interface defines rules for minting child tokens. The parent token holder registers these rules for a specific edition, identified by the edition configuration hash (editionHash). Collectors wishing to mint a child token from this edition must pass validation by successfully calling the validate function, supplying necessary information including initiator address, editionHash, and optional fulfillment data.
 */
interface IValidator {

    /**
     * @dev Initializes validator rules using the edition hash and initialization data. Decodes the data to required parameters, establishing the rules determining who can mint a copy of the edition.
     * @param editionHash The hash of the edition configuration.
     * @param initData The data bytes for initializing validation rules; parameters are encoded into bytes.
     */
    function setRules(
        bytes32 editionHash, 
        bytes calldata initData
    ) external;
    
    /**
     * @dev Validates the fulfillment of rules set by the parent token holder, using supplied data.
     * @param initiator The party initiating validation.
     * @param editionHash The hash of the edition configuration.
     * @param conditionType The type of condition for validation.
     * @param fulfillmentData Additional data required for passing validator rules.
     */
    function validate(
        address initiator, 
        bytes32 editionHash,
        uint256 conditionType,
        bytes calldata fulfillmentData
    ) external payable;

}
```
Copyedited

File: EIPS/eip-7536.md [hunk 9]
```diff
@@ -0,0 +1,226 @@
...
+        bytes32 editionHash,
+        bool isPaused
+    ) external;
+
+}
+```
+
+### The Validator Interface
+
+```
```
Comment (Pandapip1) [inline EIPS/eip-7536.md:114]:
```suggestion
```solidity
```

File: EIPS/eip-7536.md [hunk 10]
```diff
@@ -0,0 +1,226 @@
...
+
+## Specification
+
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+This standard consists of a Distributor Interface and a Validator Interface. The Distributor interface MUST be implemented by a [ERC-721](./eip-721.md) compliant contract that enables the minting of child tokens. This contract SHOULD enable any ERC-721 compliant parent token to create editions. The editions SHOULD consist of an address to a contract that implements the Validator Interface, which helps to validate the rules to mint the child token. It SHOULD also record the contract address and the tokenId of the parent token, and a set of permitted actions in the form of uint96 flags. An ERC-721-compliant contract MAY implement both the Distributor Interface and the Validator Interface, or implement only the Distributor Interface and use external contracts that implement the Validator Interface.
+
+### The Distributor Interface
+
+```
```
Comment (Pandapip1) [inline EIPS/eip-7536.md:40]:
```suggestion
```solidity
```

**Review by Pandapip1 — CHANGES_REQUESTED** [review]
Additionally: Licenses MUST be CC0-1.0, NOT MIT

**Comment by SamWilsn** [comment]
I am closing this pull request because we are in the process of separating EIPs and ERCs into distinct repositories. Unfortunately, as far as we are aware, GitHub does not provide any tools to ease this migration, so every pull request will need to be re-opened manually.

As this is a PR to create / modify an ERC, I will kindly ask you to redirect this to the new repository at [ethereum/ERCs](https://github.com/ethereum/ERCs). **We have prepared [a guide](https://github.com/ethereum/ERCs/issues/8) to help with the process.**

If there is relevant history here, please link to this PR from the new pull request.

On behalf of the EIP Editors, I apologize for this inconvenience.

**Pull request closed without merging**